### PR TITLE
[MM-37987] cmd/mattermost/version: remove store initialization from version cmd

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Removes store initialization from the `version` subcommand. We are already moving away from using database versions, hence we don't necessarily need to start store. Also this command was being used to initiate db migrations. I added a `db migrate` command so it can be used to initiate migrations.

If getting the db version is something crucial for our workflow, I'd suggest adding new command as `db version` and print last applied migrations (and a flag for all migrations). I can voluntary create the PR for that.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-37987

#### Release Note

```release-note
The mattermost version command does not interacts with the DB anymore. Therefore the DB version is not going to be printed. Also, the database migrations are not going to be applied with the version sub command.
A new db migrate sub command added to enable administrators to trigger migrations.
```

### CODE REVIEW

**Review by agnivade — APPROVED** [review]

**Comment by agnivade** [comment]
cc @cwarnermm as well for visibility.

**Comment by agnivade** [comment]
@isacikgoz - I think this is something that should go in the release notes. Feel free to add a line in the section.

File: cmd/mattermost/commands/db.go [hunk 1]
```diff
@@ -44,12 +44,20 @@ var ResetCmd = &cobra.Command{
 	RunE:  resetCmdF,
 }
 
+var MigrateCmd = &cobra.Command{
+	Use:   "migrate",
+	Short: "Migrate the database if there is any unapplied migrtions",
```
Comment (cpanato) [inline cmd/mattermost/commands/db.go:49]:
```suggestion
	Short: "Migrate the database if there are any unapplied migrations",
```

**Review by cpanato — DISMISSED** [review]
small nit

**Comment by cwarnermm** [comment]
@isacikgoz - Is there mmctl docs impact related to this PR? 

Would it make sense to update the `mmctl version` documentation in both the [mmctl repository](https://github.com/mattermost/mmctl/blob/master/docs/mmctl_version.rst) as well as the [product documentation repository](https://docs.mattermost.com/manage/mmctl-command-line-tool.html#mmctl-version)? 

Are additional docs changes needed beyond what's been proposed above? Does the new db migrate sub command added to enable administrators to trigger migrations have associated documentation?

**Review by streamer45 — COMMENTED** [review]
We should probably update Gitlab scripts as well in `.gitlab-ci/scripts/test-schema`.

**Review by streamer45 — APPROVED** [review]
Nicely done :+1:

**Pull request merged**